src/hotspot/share/runtime/mutexLocker.cpp

```diff
@@ -169,9 +169,6 @@ void assert_locked_or_safepoint(const Mutex* lock) {
   if (lock->owned_by_self()) return;
   if (SafepointSynchronize::is_at_safepoint()) return;
```

Comment:
This is actually inadequate as it would only be safe (in a lock sneaking sense) if at a safepoint AND in the vmThread. Ditto for L179.

Reply:
It is also safe for threads respecting the safepoint e.g. GC safepoint worker (safepoint cannot end until the safepoint workers stops).
In some cases we do not use Mutex during safepoint when executing with such thread.

But an entire over-haul of these asserts and what the implicit rules are is out of scope here.